mutable_buffer/src/partition.rs

```diff
@@ -156,6 +172,22 @@ impl Partition {
         }
     }
 
+    pub fn advance_chunk_state(&mut self, chunk_id: u32) {
+        if let Some(chunk) = self.closed_chunks.get(&chunk_id) {
+            chunk.advance_state();
+        } else {
+            // TODO: How to throw this exception?
+            // UnknownChunk {
+            //     partition_key: &self.key,
+            //     chunk_id,
+            // }
+            panic!(
+                "UnknownChunk (Partition key: {}, chunk id: {}",
+                &self.key, chunk_id
+            );
+        }
+    }
```

Comment:
Not sure if this will work because I'm doing it in the browser and not locally, but maybe something like this?

```suggestion
    pub fn advance_chunk_state(&mut self, chunk_id: u32) -> Result<()> {
        let _ = self.closed_chunks
            .get(&chunk_id)
            .map(|chunk| chunk.advance_state())
            .context(UnknownChunk {partition_key: &self.key, chunk_id})?; 

        Ok(())
    }
```

Reply:
Yup, it works